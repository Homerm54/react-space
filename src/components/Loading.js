import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
// import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';



const useStyle = makeStyles(theme => (
  {
    root: {
      margin: theme.spacing(4),
      height: '100%'
    },
  }
))



export default function Loading() {
  
  const classes = useStyle();
  
  
  return (
    <Grid container justify='center' alignContent='center' className={classes.root}>

      <Grid 
      item container xs={12} justify='center' 
      alignContent='center'
      >
        <CircularProgress color="secondary"/>
      </Grid>

      <Grid item>
        <Typography>Please, wait while we load everything</Typography>
      </Grid>

    </Grid>
  )
}