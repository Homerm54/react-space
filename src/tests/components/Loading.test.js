import React from 'react';
import { render } from "enzyme";
import Loading from 'components/Loading';


describe('<Loading /> Test Suit', ()=>{

  test('Should Render Correctly', ()=>{
  
    const wrapper = render(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});


