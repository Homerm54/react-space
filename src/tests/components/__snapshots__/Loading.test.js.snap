// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Test Suit Should Render Correctly 1`] = `
<div
  class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
  id="loading-component"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-content-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width:40px;height:40px"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Please, wait while we load everything...
    </p>
  </div>
</div>
`;
